src/common/app_client.rs

```diff
@@ -196,10 +211,34 @@ impl<'a> AppClient<'a> {
         };
         let body = encode_request(req)?;
 
-        let mut r = self.grpc_client.send_request(r, body)?;
+        let (mut r, headers) = self.grpc_client.send_request(r, body)?;
+        let date_str = headers
```

Comment:
would not prevent building robot / continuing in the absence of date. just set in to 0 logs would still be pushed in app